goldilocks_getter.py

```diff
@@ -457,91 +554,186 @@ def make_yaml_resource_block(soa: dict) -> str:
         A formatted string in YAML format.
     """
 
-    return f"\n{yaml.dump(soa)}\n"
+    return f"\n{yaml.dump({namespace: soa})}\n"
 
 
-def make_yaml_dump(
-    soas: pd.core.indexes.base.Index,
-    cpu_agg_req_df: pd.DataFrame,
-    mem_agg_req_df: pd.DataFrame,
-    cpu_agg_lim_df: pd.DataFrame,
-    mem_agg_lim_df: pd.DataFrame,
-    mib: bool,
+def build_resources_dict(
+    soas: tuple,
+    cpu_agg_req_df: tuple,
+    mem_agg_req_df: tuple,
+    cpu_agg_lim_df: tuple,
+    mem_agg_lim_df: tuple,
+    gib: bool,
     stat_lim: str = "max",
     stat_req: str = "median"
-) -> str:
+) -> dict:
     """Creates a single string of recommendations per SOA.
     Creates a single formatted string - not necessarily to YAML spec,
     since there are multiple SOAs on the same page - for manual input.
 
     Args:
         soas: The index from an aggregate dataframe, which is a list of SOAs.
         {cpu,mem}_agg_{req,lim}_df: Dataframes containing aggregates for resources.
-        mib: From args.mib - indicates whether to use Gi[B] or Mi[B].
+        gib: From args.gib - indicates whether to use Gi[B] or Mi[B].
         stat_{lim,req}: A str of {min, max, mean, median} to select that stat.
 
 
     Raises:
         Nothing.
 
     Returns:
-        A formatted string of all recommendations.
+        A dictionary of all current and recommended values for resources.
     """
 
     soa_recs = {}
 
-    for soa in soas:
-        soa_rec = make_resource_dict(
+    for soa in soas[0]:
+        soa_nested_dict = make_resource_dict(
             cpu_agg_req_df,
             mem_agg_req_df,
             cpu_agg_lim_df,
             mem_agg_lim_df,
             soa,
             stat_lim,
             stat_req,
-            mib
+            gib,
         )
+        soa_rec = soa_nested_dict.to_dict()
         soa_recs[soa] = soa_rec
 
-    return make_yaml_resource_block(soa_recs)
+    return soa_recs
+
+
+def export_csv(filename: str, namespace, resources: dict):
+    """Exports current, recommended and the difference in resources in a CSV file.
+    Exports current, recommended by Goldilocks and the difference in resources in
+    a CSV file that can easily be imported into spreadsheets or other data analysis
+    tools to graph the data.
+
+    Args:
+        filename: The filename that will be used to create the CSV file.
+        resources: The current and recommended resources by Goldilocks.
+
+    Raises:
+        Nothing.
+
+    Returns:
+        Nothing.
+    """
+
+    if args.gib:
+        unit = "Gi"
+    else:
+        unit = "Mi"
+
+    headers = [
+        "namespace",
+        "soa",
+        "Current [CPU] (m)",
+        "Recommendations [CPU] (m)",
+        "Difference [CPU] (m)",
+        f"Current [Memory] ({unit})",
+        f"Recommendations [Memory] ({unit})",
+        f"Difference [Memory] ({unit})"
+    ]
+
+    with open(f"{filename}.csv", 'a') as outp:
```

Comment:
Minor nitpick, but I default to double-quotes, and use single-quotes when needed for a string containing quotes.
https://www.python.org/dev/peps/pep-0008/#string-quotes